pkg/controller/identity/iamrole/controller.go

```diff
@@ -81,92 +88,97 @@ func (conn *connector) Connect(ctx context.Context, mgd resource.Managed) (manag
 	if err != nil {
 		return nil, errors.Wrap(err, errClient)
 	}
-	return &external{c}, nil
+	return &external{c, conn.client}, nil
```

Comment:
No change needed for now, but I generally prefer to include the names of fields in the struct being returned for readability and new contributors :)